src/main/java/com/github/manolo8/darkbot/gui/zones/ZoneEditor.java

```diff
@@ -85,60 +66,52 @@ public void mouseMoved(MouseEvent e) {
 
     void setup(Main main, java.util.Map<Integer, ZoneInfo> zonesByMap) {
         super.setup(main);
-        if (mapChange != null) main.mapManager.mapChange.remove(mapChange);
-        main.mapManager.mapChange.add(mapChange = map -> {
-            zoneInfo = zonesByMap.computeIfAbsent(map.id, id -> new ZoneInfo(config.BOT_SETTINGS.OTHER.ZONE_RESOLUTION));
-            ZoneEditor.this.repaint();
-        });
-        zoneInfo = zonesByMap.get(main.hero.map.id);
+
+        main.pluginAPI.requireAPI(EventBrokerAPI.class).registerListener(this);
```

Comment:
if setup is called multiple times, you will be registering to the event broker multiple times, you have to prevent that